examples/client.rs

```diff
@@ -161,9 +167,25 @@ fn main() {
 
     debug!("written {}", write);
 
+    // Create a single list of all the requests to be issued.
+    let mut reqs = std::collections::HashMap::new();
+    let mut stream_id = 0;
+
+    for url in &urls {
+        let req = format!("GET {}\r\n", url.path());
+
+        reqs.insert(stream_id, req);
+
+        stream_id += 4;
```

Comment:
Rather than associate the stream ID with the request here, we can just make a `Vec` and then calculate the stream ID based on `i` when sending them. This seems more consistent with http3-client.

Reply:
that's more or less how I started but then realised that when it comes to stream closure things were quite naiive. With H3 a lot of the mapping is done by the library but with H0.9 there is nothing, and being too simplistic with stream closures might cause some weirdness like counting a unidirectional stream closures against requests complete count.